core/src/main/java/org/apache/spark/ExecutorPluginContext.java

```diff
@@ -0,0 +1,32 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark;
+
+import com.codahale.metrics.MetricRegistry;
+import org.apache.spark.annotation.DeveloperApi;
+
+/**
+ * This class encapsulates the input for the ExecutorPlugin interface. It provides a
+ * layer of indirection to ease future extensions of executor plugins, without breaking the API.
+ */
+@DeveloperApi
+public class ExecutorPluginContext {
```

Comment:
It would be helpful to expose some other useful info here (e.g. the executor ID and the `SparkConf`).

Reply:
I have now added also "executorHostName" and "isLocal"